component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go

```diff
@@ -0,0 +1,91 @@
+// Package probabilistic_sampler provides an otelcol.processor.probabilistic_sampler component.
+package probabilistic_sampler
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/processor"
+	psp "github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+const (
+	defaultAttributeSource = "traceID"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "otelcol.processor.probabilistic_sampler",
+		Args:    Arguments{},
+		Exports: otelcol.ConsumerExports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := psp.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.processor.probabilistic_sampler component.
+
+type Arguments struct {
+	SamplingPercentage float32 `river:"sampling_percentage,attr,optional"`
+	HashSeed           uint32  `river:"hash_seed,attr,optional"`
+	AttributeSource    string  `river:"attribute_source,attr,optional"`
+	FromAttribute      string  `river:"from_attribute,attr,optional"`
+	SamplingPriority   string  `river:"sampling_priority,attr,optional"`
+
+	// Output configures where to send processed data. Required.
+	Output *otelcol.ConsumerArguments `river:"output,block"`
+}
+
+var (
+	_ processor.Arguments = Arguments{}
+)
+
+// DefaultArguments holds default settings for Arguments.
+var DefaultArguments = Arguments{
+	AttributeSource: defaultAttributeSource,
```

Comment:
Since `defaultAttributeSource` is used only once, I think it might be better to just use it here.